### PR TITLE
Implements language configurable colorizedBracketPairs property.

### LINKED ISSUE #131194: Decouple Bracket Pair Colorization From Bracket Pair Config
Not all bracket pairs should be colorized as #131003 indicates.

Language contributions should be able to provide a different list of bracket pairs to match and to colorize.

It does not make sense to colorize `<` ... `>` for html and jsx, as the color just adds noise.
However, it could be useful to declare them as bracket pair, so they are matched when clicking on them.

Keywords: IConfigurationService/LanguageConfigurationRegistry, adding bracket options to IModelService / ITextModelCreationOptions

### PR DESCRIPTION
**Pull request by hediet**
This PR fixes #131194

### CODE REVIEW

File: src/vs/workbench/contrib/codeEditor/browser/languageConfigurationExtensionPoint.ts [hunk 1]
```diff
@@ -268,6 +269,30 @@ export class LanguageConfigurationFileHandler {
...
+				console.warn(`[${languageIdentifier.language}]: language configuration: expected \`colorizedBracketPairs[${i}]\` to be an array of two strings.`);
+				continue;
+			}
+			result = result || [];
+			result.push([pair[0], pair[1]]);
+
+		}
+		return result;
+	}
+
```
Comment (hediet) [inline src/vs/workbench/contrib/codeEditor/browser/languageConfigurationExtensionPoint.ts:295]: To be honst, I really don't like this.
I can strongly recommend checking out [io-ts](https://github.com/gcanti/io-ts/blob/master/index.md#typescript-integration) or get inspired by my own library [semantic json](https://github.com/hediet/semantic-json#basic-usage).

Being able to specify deserialization logic, validation logic, json schema and the defaults in a single place (descriptively) is a huge deal. Not only here, but also for all settings.

**Comment by connor4312** [comment]
```diff
diff --git a/extensions/javascript/tags-language-configuration.json b/extensions/javascript/tags-language-configuration.json
--- extensions/javascript/tags-language-configuration.json
+++ extensions/javascript/tags-language-configuration.json
@@ -10,8 +10,9 @@
 	],
 	"colorizedBracketPairs": [
 		["{", "}"],
 		["[", "]"],
+		test
 		["(", ")"]
 	],
 	"autoClosingPairs": [
 		{ "open": "{", "close": "}" },
@@ -28,5 +29,5 @@
 		["<", ">"],
 		["'", "'"],
 		["\"", "\""]
 	]
-}
\ No newline at end of file
+}

```

**Comment by connor4312** [comment]
Ignore the above comment, I was testing something 😛

File: src/vs/workbench/contrib/codeEditor/browser/languageConfigurationExtensionPoint.ts (hunk 1 above)
Comment (alexdima) [reply src/vs/workbench/contrib/codeEditor/browser/languageConfigurationExtensionPoint.ts:295]: cc @aeschli -- would be nice if we could validate an object against a schema programmatically.

**Review by alexdima — APPROVED** [review]

**Pull request merged**